### ISSUE TITLE
Use React

### ISSUE THREAD

**Issue by oliverfei** [oliverfei/HearthstoneDeckViewer#2]

**Comment by Augustin82** [comment]
Hi. Would you be open to the UI being done in Elm?

**Comment by oliverfei** [comment]
@Augustin82 if you are offering to port everything over to use Elm and it works with all the npm packages that are being used, then yeah I don't see why not.

**Comment by Augustin82** [comment]
I'd love to start working on it some time in the very near future. I'll keep you posted if I make any progress. Thanks!

**Comment by Augustin82** [comment]
Hi! For information, I've started working on this. Hopefully I can move this along faster than it took me to start it ;)

**Comment by Augustin82** [comment]
@oliverfei I'm nearing something that could maybe kinda sorta be a decent approximation of what you currently have. I think. Do you have any sort of "test suite" or manual testing that you do, and/or a list of basic features to support, that I could lcheck against my build?

**Comment by oliverfei** [comment]
Yeah, off the top of my head, this should cover the core functionality:

- Can load multiple deck codes at once, separated by whitespace or commas
- Can load individual deck strings copied from the Hearthstone client. When loading deck strings from the client, it should also load the title.
- Can load decks from deck codes in "deckstring" url param
- Automatically updates "deckstring" url param when user adds/removes decks
- Can generate short url when user has at least one deck added
- Loaded decks:
    - Should have at least 4 fitting in a single row when the window is 1280px in width
    - Should be removable
    - Should provide users with an easy way to copy the deck code
    - Should provide users with an easy way to see full card images for cards in the deck

**Comment by Augustin82** [comment]
Thanks a lot, it's very helpful.

Off the top of my head, every item on this list should be functional (ah!) in my current version, but obviously I'll try and test it better.

I'll let you know how it goes!

**Comment by Augustin82** [comment]
Hi, some updates!
On the list you provided, everything is OK except for the following:

> Can load individual deck strings copied from the Hearthstone client. When loading deck strings from the client, it should also load the title.
I don't have a Hearthstone client with a title to test it, but doesn't that depend on whether the `deckstrings` lib can handle it?

> Can generate short url when user has at least one deck added
I haven't looked into making it work during development, and the actual production URL has CORS enabled, but this should Just Work.

> Should provide users with an easy way to copy the deck code
I need to implement visual feedback after copying the code.

I'll keep you posted!

**Comment by oliverfei** [comment]
> Can load individual deck strings copied from the Hearthstone client. When loading deck strings from the client, it should also load the title.

Yeah this is referring to the parsing logic here:
https://github.com/oliverfei/HearthstoneDeckViewer/blob/95b35a008d8d2a0a36f272fdab6339e85ae74acd/scripts/scripts.js#L80

Apologies for the lack of documentation/messy code

**Comment by Augustin82** [comment]
Thanks a lot, I saw that bit after I commented. Don't worry, I posted too quickly and should have checked thoroughly before.

**Comment by Augustin82** [comment]
That part is done! I'll work on the tooltip tomorrow. Thanks for your support!
